src/components/PopUpWithConfirmation.js

```diff
@@ -0,0 +1,42 @@
+import Popup from "./popup.js";
```

Comment:
```js

  setEventListeners() {
    const closebutton = this._popupElement.querySelector(".modal__close");

    this._popupElement.addEventListener("click", (e) => {
      if (closebutton) {
        closebutton.addEventListener("click", () => this.close());
      }
    });
```


There is a mistake in the click handler. You add a click handler inside the other click handler which causes a memory leak


It should be just this:

```js
  setEventListeners() {
    const closebutton = this._popupElement.querySelector(".modal__close");

    closebutton.addEventListener("click", () => this.close());

```